### PR TITLE
Refactor/user input controller for llm workflow

### LINKED ISSUE #169: Refactor UserInput controller for LLM workflow

### PR DESCRIPTION
**Pull request by zhiyi15**
## Once PR[#167 ](https://github.com/Linkta-org/core/pull/167) is merged, this refactor PR can be merged.
Refactored UserInputController 
Fixes #169

## Type of change

- [x]  🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨  New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] Other 🗒️ ______________

## How Has This Been Tested?

Manually tested LLM workflow with refactored controller. LLM successfully generated response and was able to store userInput into DB

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [ ] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: server/controllers/userInputController.ts
Comment (kinskiwu) [inline server/controllers/userInputController.ts:1]: @lomeintheory Please check out the changes made to this middleware.

**Review by kinskiwu — APPROVED** [review]
Thanks for the hotfix @zhiyi15 . Much appreciated! Good to go after fixing the mock user id issue.

File: mocks/userId.ts
[the remark this answers is not in the document]
Comment (zhiyi15) [reply mocks/userId.ts:1]: Agree! Haha I actually thought about it adding another mock user id instead… should’ve stick with that idea.

File: server/controllers/userInputController.ts [hunk 1]
```diff
@@ -56,7 +49,7 @@ export const fetchUserInputList = async (req: Request, res: Response) => {
   try {
     // Retrieve the user ID from the request headers or use a mock ID in non-production environments
     const userId =
-      process.env.NODE_ENV !== 'production'
+      process.env.NODE_ENV === 'production'
         ? req.headers['x-user-id'] || req.headers['x-user-id']
         : MOCK_USER_ID;
```
Comment (ChTiSh) [inline server/controllers/userInputController.ts:54]: Just trying to understand the check here, if the environment is production, then we use req.headers or else we use mock user id?

**Review by ChTiSh — APPROVED** [review]
Note that merging this PR will solve breaking changes occurred with other PR

File: server/controllers/userInputController.ts (hunk 1 above)
Comment (zhiyi15) [reply server/controllers/userInputController.ts:54]: yup correct!

**Pull request merged**